### PR TITLE
Update to not hard code svc keyvault dns suffix

### PR DESCRIPTION
**Pull request by jewzaam**
### Which issue this PR addresses:

The svc keyvault DNS suffix is hard coded to vault.azure.net and does not work in other clouds.

### What this PR does / why we need it:

Plumbs the env specific dns prefix to where it's needed.

### Test plan for issue:

Existing E2E.

### Is there any documentation that needs to be updated for this PR?

No.

### CODE REVIEW

File: pkg/deploy/generator/resources_rp.go [hunk 1]
```diff
@@ -395,6 +395,7 @@ func (g *generator) rpVMSS() *arm.Resource {
 		"rpParentDomainName",
 		"databaseAccountName",
 		"keyvaultPrefix",
+		"keyVaultDNSSuffix",
```
Comment (mjudeikis) [inline pkg/deploy/generator/resources_rp.go:398]: `template_rp.go` is missing same value

File: docs/prepare-a-shared-rp-development-environment.md [hunk 2]
```diff
@@ -336,6 +336,7 @@ locations.
    export SECRET_SA_ACCOUNT_NAME='$SECRET_SA_ACCOUNT_NAME'
    export DATABASE_ACCOUNT_NAME="\$RESOURCEGROUP"
    export KEYVAULT_PREFIX="\$RESOURCEGROUP"
+   export KEYVAULT_DNSSUFFIX="vault.azure.net"
```
Comment (mjudeikis) [inline docs/prepare-a-shared-rp-development-environment.md:339]: This goes away when wired in the template

File: hack/devtools/deploy-shared-env.sh [hunk 3]
```diff
@@ -16,6 +16,7 @@ deploy_rp_dev_predeploy() {
             "adminObjectId=$ADMIN_OBJECT_ID" \
             "fpServicePrincipalId=$(az ad sp list --filter "appId eq '$AZURE_FP_CLIENT_ID'" --query '[].objectId' -o tsv)" \
             "keyvaultPrefix=$KEYVAULT_PREFIX" \
+            "keyvaultDNSSuffix=$KEYVAULT_DNSSUFFIX" \
             "rpServicePrincipalId=$(az ad sp list --filter "appId eq '$AZURE_RP_CLIENT_ID'" --query '[].objectId' -o tsv)" >/dev/null
 }
```
Comment (mjudeikis) [inline hack/devtools/deploy-shared-env.sh:21]: This should go away

**Comment by mjudeikis** [comment]
It should not be piped from env file as we already have. 
We have this pattern where in the pre-deploy/deploy/ we inject customer variables into parameters file. We should add something like this:
```
	parameters.Parameters["keyVaultDNSSuffix"] = &arm.ParametersParameter{
		Value: d.env.Environment().KeyVaultDNSSuffix,
	}
```
In the https://github.com/Azure/ARO-RP/blob/master/pkg/deploy/deploy_rp.go#L48-L71 which will take this from Cloud Environment right to the cloud and pass it down the pipe.

I assume deploy of keyvault worked as it  ARM "appended" right value so we just need to fix RP config side. 
We should not need these values in the shell scripts at all and it should be transparent

**Comment by nilsanderselde** [comment]
Superseded by https://github.com/Azure/ARO-RP/pull/1518

**Pull request closed without merging**